### PR TITLE
lint: replace golint with revive

### PR DESCRIPTION
**Pull request by gssbzn**
## Description

`golint` has been archived and it's now deprecated on golangci-lint, they recommend to replace with revive

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -51,6 +77,7 @@ linters:
     - misspell
     - nakedret
     - nolintlint
+    - revive # Fast, configurable, extensible, flexible, and beautiful linter for Go. Drop-in replacement of golint. [fast: false, auto-fix: false]
```
Comment (threebee) [inline .golangci.yml:80]: thanks for adding this comment

File: mongodbatlas/mongodbatlas.go [hunk 2]
```diff
@@ -343,21 +344,7 @@ func SetUserAgent(ua string) ClientOpt {
...
-		return nil, err
-	}
-	var buf io.Reader
-	if body != nil {
-		if buf, err = c.newEncodedBody(body); err != nil {
-			return nil, err
-		}
-	}
-
-	req, err := http.NewRequest(method, u.String(), buf)
```
Comment (threebee) [inline mongodbatlas/mongodbatlas.go:360]: [q] For my own understanding, why was this removed?

**Review by threebee — APPROVED** [review]
LGTM, I have added a question.

File: mongodbatlas/mongodbatlas.go (hunk 2 above)
Comment (gssbzn) [reply mongodbatlas/mongodbatlas.go:360]: moved to line 408-414

Comment (gssbzn) [reply mongodbatlas/mongodbatlas.go:360]: also for context, the linter was complaining `ctx` was unused, this is an exported method so changing the signature is frowned upon so better to use it, same for `NewGZipRequest` and `NewPlainRequest`
